arangod/Graph/GraphOperations.cpp

```diff
@@ -781,10 +781,10 @@ OperationResult GraphOperations::validateEdgeVertices(
 
   // actual result doesn't matter here
   if (!resultFrom.ok()) {
-    trx.finish(resultFrom.result);
+    std::ignore = trx.finish(resultFrom.result);
```

Comment:
I would be in favor of simplifying this code here. IMO there is no need to finish the transaction here at all in case of failure.
If a transaction object goes out of scope without having been committed or aborted, it will auto-abort anyway. And read-only transactions do not need to be committed, so for them there doesn't need to be an explicit call to finish.
So we could remove the calls to `finish` here entirely, and have the one call-site handle the trx object properly (it already does).